### ISSUE TITLE
Including 3rd party Phaser Plugins 

### ISSUE THREAD

**Issue by mokargas** [rblopes/generator-phaser-plus#16]
This is more a question of approach

If I wanted to include a 3rd party plugin, installed via NPM or otherwise, what would be the best way to do that?

e.g, I've tried installing this [plugin](https://github.com/aaccurso/phaser-state-transition-plugin), but it doesn't seem to be part of the deployment process. Does it need to be included in index.html manually?

**Comment by rblopes** [comment]
For that plugin, specifically:
```js
import 'phaser-state-transition-plugin/src/phaser-state-transition-plugin';
```
Include this line in the `Preload.js` script and the initialize the plugin like explained in the repo README.

```js
this.game.stateTransition = this.game.plugins.add(Phaser.Plugin.StateTransition);
```


### tl;dr
There are two ways of including third-party libraries in game projects. The easiest is to copy the necessary scripts (and its dependencies) into `static/` and include those scripts right in the `index.html`. For libraries that depend on Phaser (like that plugin, that extends the `Phaser.Plugin` namescape), those should appear below the `<script src='phaser.js'>` tag. And, of course, no `import` is necessary after this.

The other way is using npm and importing the library package in the game code. For that to work, though, libraries must either be CommonJS compliant or, at least, [the `package.json` must declare a main script](https://docs.npmjs.com/files/package.json#main). If none of these work, a workaround is to require the specific script included in the package.

I hope that helps.

**Comment by rblopes** [comment]
To clarify, I just checked again and did a little test with the plugin and edited the first reply.
It's OK to install the plugin using npm, the problem is how that module is written.
Since it's not an CJS module, you can't just import that module to use later.
```js
import Plugin from 'phaser-state-transition-plugin'; // Does not work
```
Perhaps, this is why it didn't work for you the first time.

**Comment by mokargas** [comment]
Thanks for the explanations @rblopes 

I understand and I made a bit of an assumption about that plugin being CJS compliant. I've decided to include it manually for now. I actually might re-write it in an ES6/CJS manner
